axi4/src/axi4/bundle/ChiselType.scala

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024 Jiuyang Liu <liu@jiuyang.me>
+package org.chipsalliance.amba.axi4.bundle
+
+import chisel3.util.{Irrevocable, IrrevocableIO}
+import chisel3.{Bool, Flipped, UInt, fromIntToWidth}
+
+trait AXI4ChiselBundle extends AXI4Bundle
+
+class AW(val parameter: AXI4BundleParameter) extends AXI4ChiselBundle {
+  val id: UInt = UInt(idWidth.W)
+  val addr: UInt = UInt(addrWidth.W)
+  val len: UInt = UInt(8.W)
+  val size: UInt = UInt(3.W)
+  val burst: UInt = UInt(2.W)
+  val lock: Bool = Bool()
+  val cache: UInt = UInt(4.W)
+  val prot: UInt = UInt(3.W)
+  val qos: UInt = UInt(4.W)
+  val region: UInt = UInt(4.W)
+  val user: UInt = UInt(awUserWidth.W)
+}
+
+class W(val parameter: AXI4BundleParameter) extends AXI4ChiselBundle {
+  val data: UInt = UInt(dataWidth.W)
+  val strb: UInt = UInt((dataWidth / 8).W)
+  val last: Bool = Bool()
+  val user: UInt = UInt(wUserWidth.W)
+}
+
+class B(val parameter: AXI4BundleParameter) extends AXI4ChiselBundle {
+  val id: UInt = UInt(idWidth.W)
+  val resp: UInt = UInt(2.W)
+  val user: UInt = UInt(bUserWidth.W)
+}
+
+class AR(val parameter: AXI4BundleParameter) extends AXI4ChiselBundle {
+  val id: UInt = UInt(idWidth.W)
+  val addr: UInt = UInt(addrWidth.W)
+  val len: UInt = UInt(8.W)
+  val size: UInt = UInt(3.W)
+  val burst: UInt = UInt(2.W)
```

Comment:
For `burst` and `lock`, it is better to use enum-like defination.

Reply:
It's almost impossible to describe it in Chisel. I added the helper 7348efe2506dbdfccafe04d6692857fc4276812f